apps/managedidentity/managedidentity.go

```diff
@@ -323,9 +365,56 @@ func authResultFromToken(authParams authority.AuthParams, token accesstokens.Tok
 	return ar, err
 }
 
+// Contains checks if the element is present in the list.
+func contains[T comparable](list []T, element T) bool {
+	for _, v := range list {
+		if v == element {
+			return true
+		}
+	}
+	return false
+}
+
+// retry performs an HTTP request with retries based on the provided options.
+func retry(maxRetries int, c ops.HTTPClient, req *http.Request, s Source) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		tryCtx, tryCancel := context.WithTimeout(req.Context(), time.Second*15)
+		defer tryCancel()
+		cloneReq := req.Clone(tryCtx)
+		resp, err = c.Do(cloneReq)
+		retrylist := retryStatusCodes
+		if s == DefaultToIMDS {
+			retrylist = retryCodesForIMDS
+		}
+		if err == nil && !contains(retrylist, resp.StatusCode) {
+			return resp, nil
+		}
+		if attempt == maxRetries-1 {
```

Comment:
you can delete this block if you drain the response body at the start of the loop